File: Carpool Backend Nodejs/models/ProfileModel.js
// models/ProfileModel.js
const mongoose = require('mongoose');

const profileSchema = new mongoose.Schema({
  coverImage: String,
  profileImage: String,
  name: String,
  location: String,
  status:Boolean,
  bio: String,
  friendsList: [String],
  rides: Number,
  pickup: Number,
  user: {
    type: mongoose.Schema.ObjectId,
    ref: 'User',
  },
});

module.exports = mongoose.model('Profile', profileSchema);

File: carpool/lib/Model/profileModel.dart
class ProfileModel {
  final String coverImage;
  final String profileImage;
  final String name;
  final String location;
  final String bio;
  final List<String> friendsList;
  final int rides;
  final String pickup;

  ProfileModel({
    required this.coverImage,
    required this.profileImage,
    required this.name,
    required this.location,
    required this.bio,
    required this.friendsList,
    required this.rides,
    required this.pickup,
  });

  factory ProfileModel.fromJson(Map<String, dynamic> json) {
    return ProfileModel(
      coverImage: json['coverImage'],
      profileImage: json['profileImage'],
      name: json['name'],
      location: json['location'],
      bio: json['bio'],
      friendsList: List<String>.from(json['friendsList']),
      rides: json['rides'],
      pickup: json['pickup'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'coverImage': coverImage,
      'profileImage': profileImage,
      'name': name,
      'location': location,
      'bio': bio,
      'friendsList': friendsList,
      'rides': rides,
      'pickup': pickup,
    };
  }
}
